src/mlpack/tests/init_rules_test.cpp

```diff
@@ -126,6 +129,35 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+BOOST_AUTO_TEST_CASE(XavierInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  XavierInit<XavierUniform> xavierUniform;
+  XavierInit<XavierNormal> xavierNormal;
+  xavierUniform.Initialize(weights, 100, 100);
+  bool b = arma::all(arma::vectorise(weights) >= -(1e-2) || arma::vectorise(weights) <= (1e-2));
+  BOOST_REQUIRE_EQUAL(b, 1);
+  xavierNormal.Initialize(weights, 100, 100);
+  //std::cout << weights << std::endl;
+  auto ret = Shapiro(weights, 0.05);
```

Comment:
Honestly I don't see the need for the Shapiro Wilk test. I think that if you really wanted to test that the distribution is correct you can use the `GaussianDistribution` class and use `Train()` and ensure that the mean and variance are close to what is expected.

I realize that the Shapiro Wilk test might be a 'better' test to use here, but my concern is that the test code is itself less thoroughly tested than the `GaussianDistribution` test.